transport/RtcpReportProducer.cpp

```diff
@@ -0,0 +1,380 @@
+#include "RtcpReportProducer.h"
+#include "rtp/RtcpFeedback.h"
+#include "rtp/RtcpHeader.h"
+#include <algorithm>
+
+using namespace transport;
+
+namespace
+{
+
+bool ensurePacket(memory::UniquePacket& packet, memory::PacketPoolAllocator& allocator)
+{
+    if (!packet)
+    {
+        packet = memory::makeUniquePacket(allocator);
+    }
+
+    return !!packet;
+}
+
+// @return the removed tail span
+utils::Span<uint32_t> truncateTail(utils::Span<uint32_t>& ssrcReport, size_t n)
```

Comment:
I wonder if this should be a helper method in Span.h

Reply:
I thought about that too. But then I thought it would be easier to replace `utils::span` to `std::span` if we implement exactly the same methods (if one day we update C++ version to C++20 of curse)